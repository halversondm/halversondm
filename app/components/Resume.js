/**
 * Created by Daniel on 6/26/2016.
 */
"use strict";

import React, {Component} from "react";

class Resume extends Component {

    render() {
        return <div className="resume">
            <header className="page-header">
                <h1 className="page-title">Resume of Daniel M. Halverson</h1>
                <small><i className="fa fa-clock-o"/> Last Updated on: Sunday,
                    February 01, 2015
                </small>
            </header>
            <div className="row">
                <div
                    className="col-xs-12 col-sm-12 col-md-offset-1 col-md-10 col-lg-offset-2 col-lg-8">
                    <div className="panel panel-default">
                        <div className="panel-heading resume-heading">
                            <div className="row">
                                <div className="col-lg-12">
                                    <div className="col-xs-12 col-sm-4">
                                        <figure>
                                            <img className="img-circle img-responsive" alt=""
                                                 src="images/dan_glasses.jpeg"/>
                                        </figure>
                                        <div className="row">
                                            <div className="col-xs-12 social-btns">
                                                <div
                                                    className="col-xs-3 col-md-1 col-lg-1 social-btn-holder">
                                                    <a
                                                        href="https://plus.google.com/u/0/114718071449100470041"
                                                        className="btn btn-social btn-block btn-google"> <i
                                                        className="fa fa-google"/>
                                                    </a>
                                                </div>
                                                <div
                                                    className="col-xs-3 col-md-1 col-lg-1 social-btn-holder">
                                                    <a href="https://www.facebook.com/daniel.m.halverson"
                                                       className="btn btn-social btn-block btn-facebook">
                                                        <i
                                                            className="fa fa-facebook"/>
                                                    </a>
                                                </div>
                                                <div
                                                    className="col-xs-3 col-md-1 col-lg-1 social-btn-holder">
                                                    <a href="https://twitter.com/halversondm"
                                                       className="btn btn-social btn-block btn-twitter">
                                                        <i
                                                            className="fa fa-twitter"/>
                                                    </a>
                                                </div>
                                                <div
                                                    className="col-xs-3 col-md-1 col-lg-1 social-btn-holder">
                                                    <a href="http://www.linkedin.com/in/dmhalverson"
                                                       className="btn btn-social btn-block btn-linkedin">
                                                        <i
                                                            className="fa fa-linkedin"/>
                                                    </a>
                                                </div>
                                                <div
                                                    className="col-xs-3 col-md-1 col-lg-1 social-btn-holder">
                                                    <a href="https://github.com/halversondm"
                                                       className="btn btn-social btn-block btn-github"> <i
                                                        className="fa fa-github"/>
                                                    </a>
                                                </div>
                                                <div
                                                    className="col-xs-3 col-md-1 col-lg-1 social-btn-holder">
                                                    <a href="http://stackoverflow.com/users/975640/dan"
                                                       className="btn btn-social btn-block btn-stackoverflow">
                                                        <i className="fa fa-stack-overflow"/>
                                                    </a>
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                    <div className="col-xs-12 col-sm-8">
                                        <ul className="list-group">
                                            <li className="list-group-item">Daniel M. Halverson</li>
                                            <li className="list-group-item">Application Developer Lead
                                            </li>
                                            <li className="list-group-item">JPMorgan Chase & Co.</li>
                                            <li className="list-group-item"><i
                                                className="fa fa-phone"/>
                                                630-346-4822
                                            </li>
                                            <li className="list-group-item"><i
                                                className="fa fa-envelope"/>
                                                daniel.m.halverson@gmail.com
                                            </li>
                                        </ul>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div className="bs-callout bs-callout-danger">
                            <h4>Summary</h4>
                            <p>14 years of experience in Information Technology. Extensive
                                experience in enterprise applications through the whole SDLC
                                lifecycle.</p>
                            <p>Achieved Expert Engineer status at JPMorgan Chase in 2012.</p>
                            <p>Experience in developing Java applications in a
                                service-oriented architecture.</p>
                            <p>Experience in developing COBOL applications for batch and
                                online functions.</p>
                            <p>Experience in SQL tuning with IBM DB2.</p>
                            <p>Experience in Object Relational Mapping with Hibernate.</p>
                            <p>Experience with data transportation in the form of IBM
                                WebSphere MQ and Apache CXF / SOAP messages.</p>
                            <p>Provide on-call support with fast, well thought resolution to
                                in-the-moment problems.</p>
                            <p>Have excellent analytical, problem solving, communication and
                                interpersonal skills, with ability to interact with individuals
                                at all levels and work as a part of a team, as well as
                                independently.</p>
                        </div>
                        <div className="bs-callout bs-callout-danger">
                            <h4>Prior Experiences</h4>
                            <ul className="list-group">
                                <div className="list-group-item inactive-link">
                                    <h4 className="list-group-item-heading">Application Developer
                                        Lead at JPMorgan Chase & Co.</h4>
                                    <h5 className="list-group-item-heading">February 2013 -
                                        Current</h5>
                                    <p className="list-group-item-text">Development leader for
                                        applications in the Retail and Investment Banks. Provide
                                        direction for junior members of the team.</p>
                                    <ul>
                                        <li>Leadership of projects in an Agile methodology</li>
                                        <li>Mentor and advise others on the best design and
                                            standards for new code
                                        </li>
                                        <li>Develop the service code and unit tests.</li>
                                        <li>Instruct others on the usage of tools and procedures
                                        </li>
                                    </ul>
                                </div>
                                <div className="list-group-item inactive-link">
                                    <h4 className="list-group-item-heading">Application Developer
                                        at JPMorgan Chase & Co.</h4>
                                    <h5 className="list-group-item-heading">January 2006 -
                                        Current</h5>
                                    <p className="list-group-item-text">The CIS application is the
                                        data store for customer data across the entire JPMorgan
                                        Chase bank. Data on every customer is stored in the CIS
                                        application and every channel retrieves data from either
                                        online and/or batch functions. The CIS application is
                                        written in COBOL/CICS with JCL for batch execution.The CIS
                                        Re-engineering project overall aims to replace the legacy
                                        COBOL, platform specific application with an enterprise Java
                                        application that is platform agnostic. In this phase of the
                                        project, we implemented a pair of pilot services to test
                                        feasibility and dependability. Then, we followed with a
                                        section of ten services from our list of COBOL online
                                        functions to convert into Java.</p>
                                    <ul>
                                        <li>Leadership of projects within an enterprise release.
                                        </li>
                                        <li>Design and develop application code using COBOL, CICS,
                                            JCL and DB2
                                        </li>
                                        <li>Mentor and advise others on the best design and
                                            standards for new code
                                        </li>
                                        <li>Developed tools to assist in the productivity of team
                                            members
                                        </li>
                                        <li>Provide on-call support</li>
                                        <li>Work with business analysts to test application</li>
                                        <li>Develop testing scenarios and scripts</li>
                                        <li>Working business knowledge of the Banking Industry</li>
                                        <li>Review code for cost-savings opportunities</li>
                                        <li>Decompose existing COBOL functions to determine what
                                            business logic and SQL needed to be recreated in Java and
                                            Hibernate and deployed as a WebService.
                                        </li>
                                        <li>Develop the schema and WSDL for the Java services.</li>
                                        <li>Develop the service code and unit tests.</li>
                                        <li>Wrote technical design document with a test driven
                                            development methodology.
                                        </li>
                                        <li>Understand the deployment process and work with the
                                            configuration management team to resolve deployment
                                            issues.
                                        </li>
                                        <li>Implemented synthetic transaction monitoring using HP
                                            BPM and Nagios
                                        </li>
                                        <li>Instruct others on the usage of tools and procedures
                                        </li>
                                        <li>Worked across teams to develop JMeter scripts to test
                                            SOAP messages.
                                        </li>
                                        <li>Developed a way to use Publish/Subscribe messaging
                                            across the application server cluster to dynamically
                                            change log4j logging levels at runtime
                                        </li>
                                        <li>Developed the design pattern that the team would use for
                                            processing Point-to-Point messaging via JMS
                                        </li>
                                        <li>Developed a logging solution using Aspect Oriented
                                            Programming to consistently log fixed-byte messages
                                            entering and leaving a JMS solution
                                        </li>
                                    </ul>
                                </div>
                                <div className="list-group-item inactive-link">
                                    <h4 className="list-group-item-heading">Software Project
                                        Leader at DST Systems, Inc</h4>
                                    <h5 className="list-group-item-heading">June 2000 - December
                                        2005</h5>
                                    <p className="list-group-item-text">The TA2000 Subaccounting
                                        application was designed to provide services to
                                        broker/dealers to manage their Omnibus positions within a
                                        mutual fund. The application provides trade roll-up and
                                        triangle reconciliation. The application is written in
                                        COBOL/CICS with a thick client front-end and JCL for batch
                                        execution.</p>
                                    <ul>
                                        <li>Develop small software projects (less than 5,000 hours)
                                        </li>
                                        <li>Supported daily production cycles with fast and reliable
                                            issue resolution
                                        </li>
                                        <li>Working business knowledge of the Mutual Fund industry
                                        </li>
                                        <li>Work with several teams to build the overall project
                                        </li>
                                        <li>Develop testing scenarios and scripts</li>
                                        <li>Migrate projects through a maximum of three
                                            production-like test platforms
                                        </li>
                                        <li>Lead teams of three people to code software</li>
                                        <li>Worked jointly with a remote team to accomplish a 15,000
                                            hour project in six months
                                        </li>
                                        <li>Attended Systemation project management className</li>
                                        <li>Certified by DST in DB2 and SQL usage</li>
                                    </ul>
                                    <p/>
                                </div>
                            </ul>
                        </div>
                        <div className="bs-callout bs-callout-danger">
                            <h4>Key Expertise</h4>
                            <ul className="list-group">
                                <li className="list-group-item">Legacy Platforms including COBOL
                                    and
                                    CICS
                                </li>
                                <li className="list-group-item">Object-Oriented Analysis and
                                    Design
                                </li>
                                <li className="list-group-item">Agile Design Methodologies</li>
                                <li className="list-group-item">Legacy Platform to Object
                                    Oriented
                                    Migrations
                                </li>
                                <li className="list-group-item">Database Design and Performance
                                    Tuning
                                </li>
                                <li className="list-group-item">Messaging Platforms</li>
                                <li className="list-group-item">Web Design</li>
                            </ul>
                        </div>
                        <div className="bs-callout bs-callout-danger">
                            <h4>Language and Platform Skills</h4>
                            <ul className="list-group">
                                <div className="list-group-item inactive-link">
                                    <div className="progress">
                                        <div data-placement="top" style={{width: "80%"}}
                                             aria-valuemax="100" aria-valuemin="0"
                                             aria-valuenow="80" role="progressbar"
                                             className="progress-bar progress-bar-success">
                                            <span className="sr-only">80%</span> <span
                                            className="progress-type">Java/
                                                JavaEE/ Spring Framework </span>
                                        </div>
                                    </div>
                                    <div className="progress">
                                        <div data-placement="top" style={{width: "70%"}}
                                             aria-valuemax="100" aria-valuemin="0" aria-valuenow="1"
                                             role="progressbar"
                                             className="progress-bar progress-bar-success">
                                            <span className="sr-only">80%</span> <span
                                            className="progress-type">COBOL
                                                /CICS /JCL</span>
                                        </div>
                                    </div>
                                    <div className="progress">
                                        <div data-placement="top" style={{width: "70%"}}
                                             aria-valuemax="100" aria-valuemin="0" aria-valuenow="1"
                                             role="progressbar"
                                             className="progress-bar progress-bar-success">
                                            <span className="sr-only">70%</span> <span
                                            className="progress-type">SQL
                                            </span>
                                        </div>
                                    </div>
                                    <div className="progress">
                                        <div data-placement="top" style={{width: "65%"}}
                                             aria-valuemax="100" aria-valuemin="0" aria-valuenow="1"
                                             role="progressbar"
                                             className="progress-bar progress-bar-warning">
                                            <span className="sr-only">50%</span> <span
                                            className="progress-type">JavaScript
                                                /HTML</span>
                                        </div>
                                    </div>
                                    <div className="progress-meter">
                                        <div style={{width: "25%"}} className="meter meter-left">
                                            <span className="meter-text">I suck</span>
                                        </div>
                                        <div style={{width: "25%"}} className="meter meter-left">
                                            <span className="meter-text">I know little</span>
                                        </div>
                                        <div style={{width: "30%"}} className="meter meter-right">
                                            <span className="meter-text">I'm a guru</span>
                                        </div>
                                        <div style={{width: "20%"}} className="meter meter-right">
                                            <span className="meter-text">I''m good</span>
                                        </div>
                                    </div>
                                </div>
                            </ul>
                        </div>
                        <div className="bs-callout bs-callout-danger">
                            <h4>Education</h4>
                            <table className="table table-striped table-responsive ">
                                <thead>
                                <tr>
                                    <th>Degree</th>
                                    <th>School</th>
                                    <th>Graduation Year</th>
                                </tr>
                                </thead>
                                <tbody>
                                <tr>
                                    <td>Bachelor of Science in Business</td>
                                    <td>Eastern Illinois University</td>
                                    <td>2000</td>
                                </tr>
                                </tbody>
                            </table>
                        </div>
                    </div>
                </div>
            </div>
        </div>;
    }
}

export default Resume;

