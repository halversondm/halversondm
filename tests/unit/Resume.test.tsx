import React from "react";
import { render } from '@testing-library/react';
import Resume from "../../app/components/Resume";

test("Resume component", () => {
    const unit = <Resume />;
    const component = render(unit);
    const tree = component.container;
    expect(tree.toString()).toMatchSnapshot();
});