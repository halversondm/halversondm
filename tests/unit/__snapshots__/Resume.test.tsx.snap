// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Resume component 1`] = `"<div><div class=\\"resume\\"><header class=\\"page-header\\"><h1 class=\\"page-title\\">Resume of Daniel M. Halverson</h1><small><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"clock\\" class=\\"svg-inline--fa fa-clock \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M256 0a256 256 0 1 1 0 512A256 256 0 1 1 256 0zM232 120V256c0 8 4 15.5 10.7 20l96 64c11 7.4 25.9 4.4 33.3-6.7s4.4-25.9-6.7-33.3L280 243.2V120c0-13.3-10.7-24-24-24s-24 10.7-24 24z\\"></path></svg> Last Updated on: Friday, October 27, 2023</small></header><div class=\\"row\\"><div class=\\"col\\"><div class=\\"card\\"><div class=\\"card-header\\"><div class=\\"resume-heading\\"><div class=\\"row\\"><div class=\\"col\\"><figure><img class=\\"rounded-circle img-fluid\\" alt=\\"\\" src=\\"images/dan_suit.jpg\\"></figure></div><div class=\\"col\\"><ul class=\\"list-group\\"><li class=\\"list-group-item\\">Daniel M. Halverson</li><li class=\\"list-group-item\\">Senior Director of Software Engineering</li><li class=\\"list-group-item\\">JPMorgan Chase & Co.</li><li class=\\"list-group-item\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"phone\\" class=\\"svg-inline--fa fa-phone \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M164.9 24.6c-7.7-18.6-28-28.5-47.4-23.2l-88 24C12.1 30.2 0 46 0 64C0 311.4 200.6 512 448 512c18 0 33.8-12.1 38.6-29.5l24-88c5.3-19.4-4.6-39.7-23.2-47.4l-96-40c-16.3-6.8-35.2-2.1-46.3 11.6L304.7 368C234.3 334.7 177.3 277.7 144 207.3L193.3 167c13.7-11.2 18.4-30 11.6-46.3l-40-96z\\"></path></svg>630-346-4822</li><li class=\\"list-group-item\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"envelope\\" class=\\"svg-inline--fa fa-envelope \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z\\"></path></svg>daniel.m.halverson@gmail.com</li></ul></div></div><div class=\\"row\\"><div class=\\"col-xl-1 col-lg-1 col-md-1 col-sm-1 col-1\\"><div class=\\"social-btn-holder\\"><a href=\\"https://www.facebook.com/daniel.m.halverson\\" class=\\"btn btn-social btn-block btn-facebook\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fab\\" data-icon=\\"facebook\\" class=\\"svg-inline--fa fa-facebook \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z\\"></path></svg></a></div></div><div class=\\"col-xl-1 col-lg-1 col-md-1 col-sm-1 col-1\\"><div class=\\"social-btn-holder\\"><a href=\\"https://twitter.com/halversondm\\" class=\\"btn btn-social btn-block btn-twitter\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fab\\" data-icon=\\"twitter\\" class=\\"svg-inline--fa fa-twitter \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z\\"></path></svg></a></div></div><div class=\\"col-xl-1 col-lg-1 col-md-1 col-sm-1 col-1\\"><div class=\\"social-btn-holder\\"><a href=\\"http://www.linkedin.com/in/dmhalverson\\" class=\\"btn btn-social btn-block btn-linkedin\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fab\\" data-icon=\\"linkedin\\" class=\\"svg-inline--fa fa-linkedin \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z\\"></path></svg></a></div></div><div class=\\"col-xl-1 col-lg-1 col-md-1 col-sm-1 col-1\\"><div class=\\"social-btn-holder\\"><a href=\\"https://github.com/halversondm\\" class=\\"btn btn-social btn-block btn-github\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fab\\" data-icon=\\"github\\" class=\\"svg-inline--fa fa-github \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 496 512\\"><path fill=\\"currentColor\\" d=\\"M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z\\"></path></svg></a></div></div><div class=\\"col-xl-1 col-lg-1 col-md-1 col-sm-1 col-1\\"><div class=\\"social-btn-holder\\"><a href=\\"http://stackoverflow.com/users/975640/dan\\" class=\\"btn btn-social btn-block btn-stackoverflow\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fab\\" data-icon=\\"stack-overflow\\" class=\\"svg-inline--fa fa-stack-overflow \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 384 512\\"><path fill=\\"currentColor\\" d=\\"M290.7 311L95 269.7 86.8 309l195.7 41zm51-87L188.2 95.7l-25.5 30.8 153.5 128.3zm-31.2 39.7L129.2 179l-16.7 36.5L293.7 300zM262 32l-32 24 119.3 160.3 32-24zm20.5 328h-200v39.7h200zm39.7 80H42.7V320h-40v160h359.5V320h-40z\\"></path></svg></a></div></div></div></div></div><div class=\\"bs-callout bs-callout-danger\\"><h4>Summary</h4><p>Versatile Senior Director of Software Engineering committed to consistently finding dynamic solutions for software issues, team building, and financial discipline. Polished in finding innovative ways to efficiently build scalable, high-availability online systems. Offering a 23-year background creating and implementing software systems and leading teams.</p></div><div class=\\"bs-callout bs-callout-danger\\"><h4>Experiences</h4><ul class=\\"list-group\\"><div class=\\"list-group-item inactive-link\\"><h4 class=\\"list-group-item-heading\\">Executive Director / JPMorgan Chase & Co - Chicago, IL </h4><h5 class=\\"list-group-item-heading\\">02/2020 - Current </h5><p class=\\"list-group-item-text\\"></p><ul><li>Currently leading a team of 25 engineers across three sites for an internally developed product related to BPM / Workflow. The product has a reach of 9,000 users across the word with multiple installation patterns.</li><li>Responsible for roadmap creation with my leadership team and socialization with product owners and customers.</li><li>Designing software with team leads and architects to continue to push goals of the firm and my organization</li><li>Partnered with internal design teams to achieve the best UX experience while meeting business requirements</li><li>Mentoring direct reports for promotion and career goals</li><li>Teaching developers about the application's finer points and why decisions were made</li><li>Interviewing and hiring candidates for roles on the team</li><li>Managing to budget for runtime compute, project and associate concerns</li><li>Developing software as needed and usually in a pair programming setting</li><li>Research of products like click analytics solutions and presentation to senior leadership</li><li>Completed a modernization of a UI and API application by moving 40 microservices and 4 TB of data to the cloud</li><li>Built an SRE team from the ground up with 3 engineers</li><li>Created a development team for a legacy application starting with 4 developers and ending with 15 developers and transitioned the team to DevOps.</li><li>Drastically improved system monitoring and alerting through the usage AppDynamics, Apica, Geneos, Grafana, and Splunk</li><li>Code for Good Project Judge; a multi-city hackathon to help non-profit organizations with the technology needs via college students</li></ul></div><div class=\\"list-group-item inactive-link\\"><h4 class=\\"list-group-item-heading\\">Vice President / JPMorgan Chase & Co - Chicago, IL </h4><h5 class=\\"list-group-item-heading\\">02/2013 - 01/2020</h5><p class=\\"list-group-item-text\\"></p><ul><li>Design and develop application code using Java, JavaScript and SQL</li><li>Lead global teams up to 25 individuals through greenfield and brownfield development efforts</li><li>Experience with developing workflow applications from the ground up</li><li>Lead an effort to expand the application's infrastructure from 8 to 27 servers</li><li>Implemented the teams CI / CD solution using Jenkins and available firm tools</li><li>Developed features and resolved defects as well as provided on-call support</li><li>Lead team through code consolidation and decommission effort to reduce 3.5 MM lines of code to 2 MM</li><li>Implement software solutions on cloud using micro services patterns</li><li>Implement user interfaces in React and Angular</li><li>Promote best practices for source code through code reviews</li><li>Mentor and coach employees in their current projects and career direction</li><li>Three-time winner of company-wide hackathons and participant in Tech for Social Good</li><li>Present new technologies to senior technology management for broad understanding and inclusion</li><li>Code for Good Team Mentor; a multi-city hackathon to help non-profit organizations with the technology needs via college students</li></ul></div><div class=\\"list-group-item inactive-link\\"><h4 class=\\"list-group-item-heading\\">Associate / JPMorgan Chase & Co - Chicago, IL </h4><h5 class=\\"list-group-item-heading\\">01/2006 - 01/2013</h5><p class=\\"list-group-item-text\\"></p><ul><li>Design and develop application code using COBOL, CICS, JCL and DB2</li><li>Leadership of projects within an enterprise release</li><li>Mentor and advise others on the best design and standards for new code</li><li>Developed tools to assist in the productivity of team members</li><li>Provide on-call support</li><li>Work with business analysts to test application</li><li>Develop testing scenarios and scripts</li><li>Working business knowledge of the Banking Industry</li><li>Review code for cost-savings opportunities</li><li>Transitioned a key customer data application from COBOL to Java</li></ul></div><div class=\\"list-group-item inactive-link\\"><h4 class=\\"list-group-item-heading\\">Software Project Leader / DST Systems, Inc - Kansas City, MO</h4><h5 class=\\"list-group-item-heading\\">06/2000 - 12/2005</h5><p class=\\"list-group-item-text\\">The TA2000 Subaccounting application was designed to provide services to broker/dealers to manage their Omnibus positions within a mutual fund. The application provides trade roll-up and triangle reconciliation. The application is written in COBOL/CICS with a thick client front-end and JCL for batch execution.</p><ul><li>Develop small software projects (less than 5,000 hours)</li><li>Supported daily production cycles with fast and reliable issue resolution</li><li>Working business knowledge of the Mutual Fund industry</li><li>Work with several teams to build the overall project</li><li>Develop testing scenarios and scripts</li><li>Migrate projects through a maximum of three production-like test platforms</li><li>Lead teams of three people to code software</li><li>Worked jointly with a remote team to accomplish a 15,000 hour project in six months</li><li>Attended Systemation project management className</li><li>Certified by DST in DB2 and SQL usage</li></ul><p></p></div></ul></div><div class=\\"bs-callout bs-callout-danger\\"><h4>Skills</h4><ul class=\\"list-group\\"><li class=\\"list-group-item\\">Java, JavaScript, SQL, Oracle, Kafka, MQ, Kubernetes, Docker, Unix</li><li class=\\"list-group-item\\">Large system design with cloud technologies</li><li class=\\"list-group-item\\">Budgeting and financial management</li><li class=\\"list-group-item\\">Agile program management</li><li class=\\"list-group-item\\">Talent management and staffing</li><li class=\\"list-group-item\\">Senior Management Presentations</li></ul></div><div class=\\"bs-callout bs-callout-danger\\"><h4>Language and Platform Skills</h4><ul class=\\"list-group\\"><div class=\\"list-group-item inactive-link\\"><div class=\\"progress\\"><div data-placement=\\"top\\" style=\\"width: 80%;\\" aria-valuemax=\\"100\\" aria-valuemin=\\"0\\" aria-valuenow=\\"80\\" role=\\"progressbar\\" class=\\"progress-bar bg-success\\"><span class=\\"sr-only\\">80%</span> <span class=\\"progress-type\\">Java/ JavaEE/ Spring Framework </span></div></div><div class=\\"progress\\"><div data-placement=\\"top\\" style=\\"width: 70%;\\" aria-valuemax=\\"100\\" aria-valuemin=\\"0\\" aria-valuenow=\\"1\\" role=\\"progressbar\\" class=\\"progress-bar bg-success\\"><span class=\\"sr-only\\">80%</span> <span class=\\"progress-type\\">COBOL /CICS /JCL</span></div></div><div class=\\"progress\\"><div data-placement=\\"top\\" style=\\"width: 70%;\\" aria-valuemax=\\"100\\" aria-valuemin=\\"0\\" aria-valuenow=\\"1\\" role=\\"progressbar\\" class=\\"progress-bar bg-success\\"><span class=\\"sr-only\\">70%</span> <span class=\\"progress-type\\">SQL</span></div></div><div class=\\"progress\\"><div data-placement=\\"top\\" style=\\"width: 65%;\\" aria-valuemax=\\"100\\" aria-valuemin=\\"0\\" aria-valuenow=\\"1\\" role=\\"progressbar\\" class=\\"progress-bar bg-warning\\"><span class=\\"sr-only\\">50%</span> <span class=\\"progress-type\\">JavaScript /HTML</span></div></div><div class=\\"progress-meter\\"><div style=\\"width: 25%;\\" class=\\"meter meter-left\\"><span class=\\"meter-text\\">I suck</span></div><div style=\\"width: 25%;\\" class=\\"meter meter-left\\"><span class=\\"meter-text\\">I know little</span></div><div style=\\"width: 30%;\\" class=\\"meter meter-right\\"><span class=\\"meter-text\\">I'm a guru</span></div><div style=\\"width: 20%;\\" class=\\"meter meter-right\\"><span class=\\"meter-text\\">I'm good</span></div></div></div></ul></div><div class=\\"bs-callout bs-callout-danger\\"><h4>Education</h4><table class=\\"table table-striped table-responsive \\"><thead><tr><th>Degree</th><th>School</th><th>Graduation Year</th></tr></thead><tbody><tr><td>Bachelor of Science in Business</td><td>Eastern Illinois University</td><td>2000</td></tr></tbody></table></div></div></div></div></div></div>"`;
